File: Version2/GO/ReadCurrenciesRestApi.go
/**
 * @version GO
 * @author MARIO SAM <eu@mariosam.com.br>
 * @see I would love to work with you instead solving web code tests: hire me!
 */
package GO

import (
	"encoding/json"
	"io/ioutil"
	"net/http"
)

/**
 * Simple example of REST API call to return all available currencies of Magento store.
 */
func getCurrenciesWithREST(site string) []string {
	results := make([]string, 0)
	response, _ := http.Get(site + "/rest/V1/directory/currency")

	if response.StatusCode == 200 {
		defer response.Body.Close()
		contents, _ := ioutil.ReadAll(response.Body)

		var obj = make(map[string][]string)
		json.Unmarshal(contents, &obj)

		results = obj["available_currency_codes"]
	}

	return results
}

File: Version2/PHP/ReadCurrenciesRestApi.php
<?php
/**
 * @version PHP
 * @author MARIO SAM <eu@mariosam.com.br>
 * @see I would love to work with you: hire me!
 */
namespace PHP;

class ReadCurrenciesRestApi {

    /**
     * Simple example of REST API call to return all available currencies of Magento store.
     */
    public function getCurrenciesWithREST( $site = "" ) {
        $results = array();

        try {
            $ch = curl_init( $site . "/rest/V1/directory/currency" );
            curl_setopt( $ch, CURLOPT_CUSTOMREQUEST, "GET" );
            curl_setopt( $ch, CURLOPT_RETURNTRANSFER, true );
                
            $json = curl_exec( $ch );
            $httpcode = curl_getinfo( $ch, CURLINFO_HTTP_CODE );
            curl_close( $ch );

            if ( $httpcode == 200 ) {
                $obj = json_decode( $json );

                $results = $obj->available_currency_codes;
            }
        } catch ( \Exception $e ) {
            echo "Error: " . $e->getMessage();
        }

        return $results;
    }
}
